src/Lighthouses.java

```diff
@@ -0,0 +1,170 @@
+    }
+
+    public void putShips( char[][] tab ) {
+        boolean check = false;
+        System.out.println("Podaj koordynaty kom�rki w kt�r� chcesz wstawi� statek:");
+        System.out.println("Wpisz 'q' aby zako�czy� gr�, 'z' aby zdj�� statek, 's' aby sprawdzi� plansz�\n'd'aby i�� dalej");
+        char quit = key.next().charAt(0);
+        if (quit == 'z') {
+            removeShip(board);
+        } else if (quit == 's') {
+            checkLighthouses(board);
+            putShips(board);
+        } else if (quit == 'q') {
+            System.exit(0);
+        }
+        else if(quit=='d') {
+            int indexNumberRow;
+            int indexNumberColumn;
+            try {
+                System.out.print("Rz�d:");
+                indexNumberRow = key.nextInt();
+                System.out.print("Kolumna:");
+                indexNumberColumn = key.nextInt();
+                int j;
+                for (int i = indexNumberRow - 1; i <= indexNumberRow + 1; i++) {
+                    for (j = indexNumberColumn - 1; j <= indexNumberColumn + 1; j++) {
+                        /*System.out.print(tab[i][j] + " ");*/
+                        if (tab[i][j] == '_') {
+                            check = true;
+                        } else {
+                            System.out.println("Spr�buj jeszcze raz:");
+                            printBoard(board);
+                            check = false;
+                            putShips(board);
+                        }
+                    }
+                }
+                if (check == true) {
+                    tab[indexNumberRow][indexNumberColumn] = 'x';
+                    printBoard(board);
+                    putShips(board);
+                }
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("Poza zakresem tablicy, spr�buj jeszcze raz");
+                putShips(board);
+            }
+        }
+    }
+
+    public void removeShip( char[][] tab ) {
+        System.out.println("Zdejmij statek, podaj rz�d i kolumn�:");
+        int indexNumberRow;
+        int indexNumberColumn;
+        try {
+            System.out.println("Rz�d:");
+            indexNumberRow = key.nextInt();
+            System.out.println("Kolumna:");
+            indexNumberColumn = key.nextInt();
+            if (tab[indexNumberRow][indexNumberColumn] == 'x') {
+                board[indexNumberRow][indexNumberColumn] = '_';
+                System.out.println("Zdj��em statek z:[" + indexNumberRow + "] [" + indexNumberColumn);
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Poza zakresem tablicy, spr�buj jeszcze raz");
+            removeShip(board);
+        }
+        putShips(board);
+    }
+
+    /* ta metoda musi por�wnywa� czy dla ka�dej latarni zgadza si� liczba uzyskana z policzenia x-�w*/
+    public void  checkLighthouses( char[][] tab ) {
+        boolean winOrNot = false;
```

Comment:
winOrNot ?

przy definiowaniu zmiennych boolean wybierz albo przypadek pozytywny albo negatywny. Zazwyczaj używamy pozytywnych. Czyli... np.

boolean win;

win = true (oznacza że wygrał)
win = false (oznacza, że przegrał)

gdy tak opiszesz, to warunek typu

if(win) { //   "Jeśli wygrał? "   bardzo fajnie się czyta prawda ?

}

albo

if(!win) {  // "Jeśli nie wygrał?"

}

--------------
jak to wygląda teraz ?

if(winOrNot) {   // "Jeśli wygrał albo nie wygrał?"   // śmiesznie prawda ? :D

}